packages/aws-cdk-lib/aws-stepfunctions/lib/fields.ts

```diff
@@ -556,23 +562,23 @@ function validateJsonPath(path: string) {
   ];
   const intrinsicFunctionFullNames = intrinsicFunctionNames.map((fn) => `States.${fn}`);
   if (path !== '$'
-    && !path.startsWith('$.')
```

Comment:
Is this intended change?

Reply:
Yes, of course. Until now, JSONPath only supported paths starting with $. or $$. (e.g. $.foo.bar). If any other path is specified, a validation error is thrown.

Variables, on the other hand, start with $ without a dot (e.g. $foo.bar). Without this change, you will be left unable to specify variables in JSONPath.